Dockerfile

```diff
@@ -1,45 +1,75 @@
-FROM alpine:3.20 AS builder
+ARG BUILD_OS
+ARG BUILD_BASE
 
-RUN apk add --update --no-cache \
+FROM ${BUILD_OS}:${BUILD_BASE} AS builder
```

Comment:
I'm a bit unsure about this. Perhaps it would make more sense to separate the docker files instead, so that a by-the-book docker build works?

E.g.,

```
docker/alpine/Dockerfile
docker/debian/Dockerfile
```

?

Reply:
Yes but it is much more work to maintain than just two build arguments
Don't you think ?

Reply:
OK, I'm fine with your variant